openmdao/test_suite/components/branin.py

```diff
@@ -0,0 +1,74 @@
+"""
+Test objects for the single-discipline Branin (or Branin-Hoo) problem.
+
+The Branin, or Branin-Hoo, function has three global minima. The recommended
+values of a, b, c, r, s and t are:
+  a = 1
+  b = 5.1/(4pi2),
+  c = 5/pi,
+  r = 6,
+  s = 10 and
+  t = 1/(8pi).
+
+This function is usually evaluated on the square x0 ~ [-5, 10], x1 ~ [0, 15].
+
+The global minimum can be found at f(x) = 0.397887
+"""
+import numpy as np
+
+from openmdao.api import ExplicitComponent
+
+
+class Branin(ExplicitComponent):
+    """
+    The Branin test problem. This version is the standard version and
+    contains two continuous parameters.
+    """
+
+    def setup(self):
+        """
+        Define the independent variables, output variables, and partials.
+        """
+        # Inputs
+        self.add_input('x0', 0.0)
+        self.add_input('x1', 0.0)
+
+        # Outputs
+        self.add_output('f', val=0.0)
+
+        self.declare_partials(of='f', wrt=['x0', 'x1'])
+
+    def compute(self, inputs, outputs):
+        """
+        Define the function f(xI, xC).
+
+        Here x0 is integer and x1 is continuous.
```

Comment:
This comment seems to conflict with the class docstring, and the definition of x0 and x1 as float 0.0... was this comment meant to be attached to the example problem rather than to the class?

Reply:
Yeah, i could clean this up a little.